samcli/local/lambda_service/service.py

```diff
@@ -0,0 +1,103 @@
+"""Local Lambda Service"""
+
+import logging
+import os
+
+from flask import Flask, jsonify
+
+LOG = logging.getLogger(__name__)
+
+
+class LocalLambdaService(object):
+
+    _DEFAULT_PORT = 3001
+    _DEFAULT_HOST = '127.0.0.1'
+
+    def __init__(self, function_name_list, lambda_runner, port=None, host=None, stderr=None):
+        """
+        Creates a Local Lambda Service
+
+        Parameters
+        ----------
+        function_name_list list of str
+            List of the Function Logical Ids
+        lambda_runner samcli.commands.local.lib.local_lambda.LocalLambdaRunner
+            The Lambda runner class capable of invoking the function
+        port int
+            Optional. port for the service to start listening on. Defaults to 3001
+        host str
+            Optional. host to start the service on Defaults to '127.0.0.1'
+        stderr io.BaseIO
+            Optional stream where the stderr from Docker container should be written to
+        """
+        self.function_name_list = function_name_list
+        self.lambda_runner = lambda_runner
+        self.port = port or self._DEFAULT_PORT
+        self.host = host or self._DEFAULT_HOST
+        self._app = None
+        self.stderr = stderr
+
+    def create(self):
+        """
+        Creates a Flask Application that can be started.
+        """
+        self._app = Flask(__name__)
+
+        for function_name in self.function_name_list:
+            path = '/2015-03-31/functions/{}/invocations'.format(function_name)
```

Comment:
At the moment, I create a path for each function but not sure this is the right approach at the moment. The version of the API is in the path, so if a new one comes out we have to then update. Instead, we can catch all paths and do some path matching (we have to do some of this already to get the Function Logical Id out of the path). If we do the path matching, we can ignore the first part (2015-03-31). This would make our code a little more future proof but we have do handle most of the error cases and route handling ourselves (and not get some of this from Flask). 

I am leaning towards accepting all paths and doing the path matching/inspection but haven't fully convinced myself yet. Thoughts?

Reply:
Invoke might be version locked today, but could change in the future. Since we only support one Invoke API, this is good for now. It doesn't lead us thru one way doors and can be changed easily in the future.